fhir-core/src/main/java/com/ibm/fhir/core/util/URLSupport.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * (C) Copyright IBM Corp. 2021
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.ibm.fhir.core.util;
+
+import static com.ibm.fhir.core.util.LRUCache.createLRUCache;
+
+import java.io.UnsupportedEncodingException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.net.URLDecoder;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+/**
+ * A utility class for working with URLs
+ */
+public class URLSupport {
+    private static final Map<String, URL> URL_CACHE = createLRUCache(128);
+
+    private URLSupport() { }
+
+    public static String decode(String s) {
```

Comment:
would be nice to have method prologues for these methods, possibly with simple example of an input string and what's returned for the less obvious ones

Reply:
Will add some Javadoc

Reply:
Javadoc added